### PR TITLE
makedirs can fail if not setting exist_ok=True

### PR DESCRIPTION
**Pull request by bmaranville**
I'm sure it was a weird edge-case race condition, but this code failed for me when working with a library that was using sasmodels.  I think the error can be avoided by setting "exist_ok=True" in makedirs

```
  File "/home/bbm/dev/sasmodels/sasmodels/core.py", line 132, in load_model
    return build_model(load_model_info(model_name),
  File "/home/bbm/dev/sasmodels/sasmodels/core.py", line 339, in build_model
    return kerneldll.load_dll(source['dll'], model_info, numpy_dtype)
  File "/home/bbm/dev/sasmodels/sasmodels/kerneldll.py", line 301, in load_dll
    filename = make_dll(source, model_info, dtype=dtype)
  File "/home/bbm/dev/sasmodels/sasmodels/kerneldll.py", line 272, in make_dll
    os.makedirs(SAS_DLL_PATH)
  File "/home/bbm/miniforge3/envs/saswebcalc/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/home/bbm/.sasmodels/compiled_models'
```

### CODE REVIEW

**Comment by butlerpd** [comment]
@krzywon to verify if it is ready to merge.

**Comment by bmaranville** [comment]
Somehow I screwed up this trivial merge.  The web merge tool is not very good.  DO NOT ACCEPT THIS PR.

**Comment by bmaranville** [comment]
Ok, someone else changed the function signature and the body of the function in addition to this very simple change, which is causing the merge conflict.  I am not able to make the merge cleanly for this reason, and I know nothing of the intent of the other changes.

File: sasmodels/kerneldll.py [hunk 1]
```diff
@@ -272,9 +272,11 @@ def make_dll(source, model_info, dtype=F64, system=False):
...
 
     if not os.path.exists(dll):
-        # Make sure the DLL path exists.
-        if not os.path.exists(SAS_DLL_PATH):
-            os.makedirs(SAS_DLL_PATH)
+        # Make sure the DLL path exists. Use abspath since python docs warn
+        # that makedirs is not robust against '..' in path.
+        os.makedirs(os.path.abspath(SAS_DLL_PATH), exist_ok=True)
+        basename = splitext(os.path.basename(dll))[0] + "_"
+        system_fd, filename = tempfile.mkstemp(suffix=".c", prefix=basename)
```
Comment (pkienzle) [inline sasmodels/kerneldll.py:279]: Merge mess: these two lines are duplicated in `if not system` below.

The purpose of the @llimeht change was to create precompiled .so files for the debian distribution. I've asked him to review your change, though I think just deleting these two lines will be fine.

**Pull request closed without merging**